### PR TITLE
Document sentence triggers

### PR DESCRIPTION
**Pull request by synesthesiam**
## Proposed change

Document sentence triggers using Assist.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/94613
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/automation/trigger.markdown [hunk 1]
```diff
@@ -892,6 +892,32 @@ automation:
 See the [Calendar](/integrations/calendar/) integration for more details on event triggers and the
 additional event data available for use by an automation.
 
+## Sentence trigger
+
+A sentence trigger fires when [Assist](https://www.home-assistant.io/voice_control/) matches a sentence from a voice assistant, such as [ESPHome](https://www.home-assistant.io/voice_control/thirteen-usd-voice-remote/) or an [analog phone](https://www.home-assistant.io/voice_control/worlds-most-private-voice-assistant/).
```
Comment (balloob) [inline source/_docs/automation/trigger.markdown]: I wouldn’t mention implementations of Assist. Leave that up to the page that Assist links to. I would mention that it requires the Home Assistant conversation agent.

File: source/_docs/automation/trigger.markdown [hunk 2]
```diff
@@ -892,6 +892,32 @@ automation:
...
+
+Sentences are allowed to use some basic [template syntax](https://developers.home-assistant.io/docs/voice/intent-recognition/template-sentence-syntax/#sentence-templates-syntax) like optional and alternative words. For example, "[it's] party time" will match "party time" and "it's party time".
```
Comment (balloob) [inline source/_docs/automation/trigger.markdown]:
```suggestion
Sentences are allowed to use some basic [template syntax](https://developers.home-assistant.io/docs/voice/intent-recognition/template-sentence-syntax/#sentence-templates-syntax) like optional and alternative words. For example, "[it's ]party time" will match "party time" and "it's party time".
```

File: source/_docs/automation/trigger.markdown [hunk 3]
```diff
@@ -892,6 +892,32 @@ automation:
...
+The sentences matched by this trigger will be:
+
+- party time
+- it's party time
+- happy new year
+- happy birthday
+
+Punctuation and casing are ignored, so "It's PARTY TIME!!!" will also match.
+
+Different voice assistants may return different text for numbers, dates, times, or unusual words. You may need to use [Assist's debug view](https://www.home-assistant.io/voice_control/troubleshooting/) to determine which sentences to include in your trigger.
```
Comment (balloob) [inline source/_docs/automation/trigger.markdown]: This paragraph causes more confusion than that it helps. We also don’t do responses yet, so I would just remove it.

File: source/_docs/automation/trigger.markdown (hunk 1 above)
Comment (synesthesiam) [reply source/_docs/automation/trigger.markdown]: Linking to the conversation integration now

**Review by balloob — APPROVED** [review]

**Pull request merged**